non-matching tags are not ignored

If I give a `tag_regex`, I would expect non-matching tags to be ignored, instead they seem to fail:
```
$ python setup.py sdist
/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'long_description_content_type'
  warnings.warn(msg)
/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools_scm/version.py:92: UserWarning: tag 'apache-arrow-js-0.9.9' no version found
  warnings.warn("tag %r no version found" % (tag,))
Traceback (most recent call last):
  File "setup.py", line 557, in <module>
    url="https://arrow.apache.org/"
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/distutils/core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools/dist.py", line 333, in __init__
    _Distribution.__init__(self, attrs)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/distutils/dist.py", line 281, in __init__
    self.finalize_options()
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools/dist.py", line 476, in finalize_options
    ep.load()(self, ep.name, value)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools_scm/integration.py", line 23, in version_keyword
    dist.metadata.version = get_version(**value)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools_scm/__init__.py", line 135, in get_version
    parsed_version = _do_parse(config)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools_scm/__init__.py", line 88, in _do_parse
    config, "setuptools_scm.parse_scm"
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools_scm/__init__.py", line 45, in _version_from_entrypoint
    version = _call_entrypoint_fn(config, ep.load())
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools_scm/__init__.py", line 33, in _call_entrypoint_fn
    return fn(config.absolute_root, config=config)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools_scm/git.py", line 134, in parse
    return meta(tag, config=config, node=node, dirty=dirty, branch=branch)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.6/site-packages/setuptools_scm/version.py", line 196, in meta
    assert parsed_version is not None, "cant parse version %s" % tag
AssertionError: cant parse version apache-arrow-js-0.9.9
```

In this case my options are as follows:
```
    use_scm_version={"root": "..",
                     "relative_to": __file__,
                     "tag_regex": '^apache-arrow-([\.0-9]+)$'},
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -4,6 +4,8 @@ v3.1.0
 * fix #297 - correct the invocation in version_from_scm and deprecate it as its exposed by accident
 * fix #298 - handle git file listing on empty repositories
 * fix #268 - deprecate ScmVersion.extra
+* fix #303 and #283 by adding the option `git_describe_command` to allow the user to control the
+way that `git describe` is called.
 
 v3.0.6
 ======

--- a/README.rst
+++ b/README.rst
@@ -210,6 +210,13 @@ The currently supported configuration keys are:
   defaults to the value of ``setuptools_scm.config.DEFAULT_TAG_REGEX``
   (see `config.py <src/setuptools_scm/config.py>`_).
 
+:git_describe_command:
+  This command will be used instead the default `git describe` command.
+  Use with caution, this is a function for advanced use, and you should be
+  familiar with the setuptools_scm internals to use it.
+
+  The default value is set by ``setuptools_scm.git.DEFAULT_DESCRIBE``
+  (see `git.py <src/setuptools_scm/git.py>`_).
 
 To use setuptools_scm in other Python code you can use the
 ``get_version`` function:

--- a/src/setuptools_scm/__init__.py
+++ b/src/setuptools_scm/__init__.py
@@ -121,6 +121,7 @@ def get_version(
     relative_to=None,
     tag_regex=None,
     parse=None,
+    git_describe_command=None,
 ):
     """
     If supplied, relative_to should be a file from which root may
@@ -138,6 +139,7 @@ def get_version(
     config.relative_to = relative_to
     config.tag_regex = tag_regex
     config.parse = parse
+    config.git_describe_command = git_describe_command
 
     parsed_version = _do_parse(config)
 

--- a/src/setuptools_scm/config.py
+++ b/src/setuptools_scm/config.py
@@ -61,6 +61,7 @@ class Configuration(object):
         self.write_to_template = None
         self.parse = None
         self.tag_regex = DEFAULT_TAG_REGEX
+        self.git_describe_command = None
 
     @property
     def absolute_root(self):

--- a/src/setuptools_scm/git.py
+++ b/src/setuptools_scm/git.py
@@ -101,9 +101,12 @@ def parse(
     if pre_parse:
         pre_parse(wd)
 
+    if config.git_describe_command:
+        describe_command = config.git_describe_command
+
     out, unused_err, ret = wd.do_ex(describe_command)
     if ret:
-        # If 'git describe' failed, try to get the information otherwise.
+        # If 'git git_describe_command' failed, try to get the information otherwise.
         rev_node = wd.node()
         dirty = wd.is_dirty()
 

--- a/testing/test_git.py
+++ b/testing/test_git.py
@@ -191,3 +191,18 @@ def test_git_feature_branch_increments_major(wd):
     wd("git checkout -b feature/fun")
     wd.commit_testfile()
     assert wd.get_version(version_scheme="python-simplified-semver").startswith("1.1.0")
+
+
+@pytest.mark.issue("https://github.com/pypa/setuptools_scm/issues/303")
+def test_not_matching_tags(wd):
+    wd.commit_testfile()
+    wd("git tag apache-arrow-0.11.1")
+    wd.commit_testfile()
+    wd("git tag apache-arrow-js-0.9.9")
+    wd.commit_testfile()
+    assert wd.get_version(
+        tag_regex=r"^apache-arrow-([\.0-9]+)$",
+        git_describe_command="git describe --dirty --tags --long --exclude *js* ",
+    ).startswith(
+        "0.11.2"
+    )

